api/handlers/service_binding_test.go

```diff
@@ -72,476 +72,555 @@ var _ = Describe("ServiceBinding", func() {
-				Expect(createServiceBindingMessage.ServiceInstanceGUID).To(Equal("service-instance-guid"))
-				Expect(createServiceBindingMessage.SpaceGUID).To(Equal("space-guid"))
-				Expect(rr).To(HaveHTTPStatus(http.StatusAccepted))
-				Expect(rr).To(HaveHTTPHeaderWithValue("Location",
-					ContainSubstring("/v3/jobs/managed_service_binding.create~service-binding-guid")))
+				Expect(actualAppGUID).To(Equal("app-guid"))
 			})
 
-			When("creating the ServiceBinding errors", func() {
+			When("getting the app is forbidden", func() {
 				BeforeEach(func() {
-					serviceBindingRepo.CreateServiceBindingReturns(repositories.ServiceBindingRecord{}, errors.New("boom"))
+					appRepo.GetAppReturns(repositories.AppRecord{}, apierrors.NewForbiddenError(nil, repositories.ServiceBindingResourceType))
 				})
 
-				It("returns an error", func() {
-					expectUnknownError()
+				It("returns a not found error", func() {
+					expectNotFoundError(repositories.ServiceBindingResourceType)
 				})
 			})
-		})
-	})
 
-	Describe("GET /v3/service_credential_bindings/{guid}", func() {
-		BeforeEach(func() {
-			requestMethod = http.MethodGet
-			requestPath = "/v3/service_credential_bindings/service-binding-guid"
-			requestBody = ""
-		})
-
-		It("returns the service binding", func() {
-			Expect(rr).To(HaveHTTPStatus(http.StatusOK))
-			Expect(rr).To(HaveHTTPHeaderWithValue("Content-Type", "application/json"))
-			Expect(rr).To(HaveHTTPBody(SatisfyAll(
-				MatchJSONPath("$.guid", "service-binding-guid"),
-				MatchJSONPath("$.links.self.href", "https://api.example.org/v3/service_credential_bindings/service-binding-guid"),
-			)))
-		})
+			When("getting the App errors", func() {
+				BeforeEach(func() {
+					appRepo.GetAppReturns(repositories.AppRecord{}, errors.New("boom"))
+				})
 
-		When("the service bindding repo returns an error", func() {
-			BeforeEach(func() {
-				serviceBindingRepo.GetServiceBindingReturns(repositories.ServiceBindingRecord{}, errors.New("get-service-binding-error"))
+				It("returns an error and doesn't create the ServiceBinding", func() {
+					expectUnknownError()
+					Expect(serviceBindingRepo.CreateServiceBindingCallCount()).To(Equal(0))
+				})
 			})
 
-			It("returns an error", func() {
-				expectUnknownError()
+			When("the App and the ServiceInstance are in different spaces", func() {
+				BeforeEach(func() {
+					appRepo.GetAppReturns(repositories.AppRecord{SpaceGUID: spaceGUID}, nil)
+					serviceInstanceRepo.GetServiceInstanceReturns(repositories.ServiceInstanceRecord{SpaceGUID: "another-space-guid"}, nil)
+				})
+
+				It("returns an error and doesn't create the ServiceBinding", func() {
+					expectUnprocessableEntityError("The service instance and the app are in different spaces")
+					Expect(serviceBindingRepo.CreateServiceBindingCallCount()).To(Equal(0))
+				})
 			})
 		})
 
-		When("the user is not authorized", func() {
+		Describe("GET /v3/service_credential_bindings/{guid}", func() {
```

Comment:
The PR moves `Describe("GET /v3/service_credential_bindings/{guid}"` under `Describe("POST /v3/service_credential_bindings"` which is not correct. The `Describe` should remain on the same level as `Describe("POST /v3/service_credential_bindings"`